### ISSUE TITLE
VelocityMacro.postman_collection.json

### ISSUE THREAD

**Issue by rashik1144** [dotCMS/core#27268]

**Comment by mbiuki** [comment]
@rashik1144  - when you upload a non .vtl macro, does it avoid and give proper notification? If not, we should add this as a feature - @bryanboza

**Comment by rashik1144** [comment]
@mbiuki Yes. 500 error is given when the test fails but the postman shows that the test code is throwing error when you upload wrong file. 

![Image](https://github.com/dotCMS/core/assets/128124382/c5989b51-1297-45e5-954e-3d0fab31e656)



![Image](https://github.com/dotCMS/core/assets/128124382/3553ba9c-8f9d-409a-9e6d-a45c824c38e8)

**Comment by mbiuki** [comment]
It should be a 40x msg not 500 - this requires investigation

**Comment by sfreudenthaler** [comment]
From our last security meeting.  Not prioritizing this because the impact is low.  But it's something that someone new could pickup or we could pickup with a few spare cycles.

Here are some error codes we could use [source:](https://stackoverflow.com/questions/70010742/appropriate-http-status-code-for-file-upload)

> 400 - argument to return 400 bad request is that request failed due to validation. Argument to not to use is sometime it also indicates syntactical error while making the request.
> 415 - argument to return 415 Unsupported Media is that file extension is not same. (Considering Media as file), but content type (multipart/form-data) matches and is supported.
> 422 - Argument to use 422 unprocessable entity is that server understands the content-type (multipart/form-data) but still can't process the request. But other though is, it can still process if the use case is only to upload the file.

Based on what I see in the Mozilla Developer Network web docs the most appropriate is `415 Unsupported Media Type`

**Comment by mbiuki** [comment]
In reality, we should not receive any 50x error codes which would mean the server encountered an unexpected condition that prevented it from fulfilling the request. Usually 40x and 20x are the prominent healthy codes that we should support for our sites.
